ONSdigital/eq-author-app
# 1507 implement firestore layer

### What is the context of this PR?

This task involved creating a new datastore so that the API can interact and store data within Firestore instead of DynamoDB. Functionally, Author (front end) has not changed as part of this.

This change does not require a migration, however in future we will need to decide upon a strategy for moving questionnaires from DynamoDB into Firestore.

### How to review 

Ensure that the `DATABASE` environment variable within the `docker-compose` file is set to `firestore`. This will change the database from DynamoDB to Firestore and allow the right datastore file to be used.

* Import the **Capability examples** questionnaire from pre-prod Author into your local environment and:
    * ensure it can be opened in Author;
    * then, ensure it can be viewed in Runner by pressing the **view survey** button
* Does this require a migration? We need one if existing JSON schema properties change

### What to do after everything is green

1. * [ ] Bring this branch up-to-date with Origin/Master
2. * [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. * [ ] Click the **Merge** button on this pull request
4. * [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. * [ ] Move the Trello card for this task into the next stage of the process
